101-vm-ws2016-visualstudio2019/GettingStarted-wsvs.md

```diff
@@ -0,0 +1,195 @@
+
+The only parameters that we need to inform are:  **adminUsername**, **adminPassword** and **vmName**.
+
+Don't worry about changing anything on the file, either on the portal or using Azure CLI, you gonna be asked to insert this information, but bear in mind that there is some requirement for those two parameters:
+
+
+- *adminUsername:* Usernames can be a maximum of 20 characters in length and cannot end in a period (".").
+
+The following usernames are not allowed:  
+
+  | | ||
+:--------- | :-------- |:-----------    |:------
+administrator     | admin | user     |user1|
+test|     user2|     test1|     user3|
+admin1     |1 |    123|     a|
+actuser     |adm     |admin2|     aspnet
+backup     |console |    david |    guest
+john     |owner     |root |    server
+sql     |support     |support_388945a0     |sys
+test2     |test3     |user4     |user5
+
+
+- *adminPassword* There are varying password length requirements, depending on the tool you are using:
+
+    Portal - between 12 - 72 characters  
+    PowerShell - between 8 - 123 characters  
+    CLI - between 12 - 123  
+
+    - Have lower characters
+    - Have upper characters
+    - Have a digit
+    - Have a special character (Regex match [\W_])
+
+. The following passwords are not allowed:
+
+>abc@123     iloveyou!     P@$$w0rd     P@ssw0rd     P@ssword123
+Pa$$word     pass@word1     Password!     Password1     Password22
+
+- *vmName:* The name of your Virtual Machine. Keep in mind that the Virtual Machine Name is key for you to deploy multiple resources in the same group. It will come with a default value, but you can change it when deploying your resource. Also, know that the maximum characters allowed are 15.
+
+Keeping that in mind, let's rock with the Deployment.
+
+##Deployment
+There are a few ways to deploy this template.
+You can use [PowerShell](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-deploy), [Azure CLI](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-deploy-cli), [Azure Portal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-deploy-portal) or your favorite SDK.
+
+For this task, we will deploy using the Portal and Azure CLI, I choose to use the Visual Code with Azure CLI extensions, if you like, you can find more information [here](https://code.visualstudio.com/docs/azure/extensions). But bare in mind that you don't need to use the Visual Code, you can stick with the old good always present **Command Line** on Windows or any **bash terminal**.
+
+###Using Azure CLI with Visual Code
+type on the terminal windows: **az login**
+
+![Screen](./images/azlogin.png)
+
+You gonna be redirected to the Azure Portal where you can use your credentials to log in.
```

Comment:
more grammar - probably good to have someone edit/review in detail

Reply:
Hey Brian, thanks for all the feedback. Sorry for the silly question, just because it's bugging me. When you mean grammar, you meant English grammar or markdown grammar? (if this exists ;-)

Reply:
English Grammar... Markdown Syntax ;)